cyraft/node.py

```diff
@@ -661,11 +696,19 @@ def _reset_election_timeout(self) -> None:
         assert self._state == RaftState.FOLLOWER, "Only followers should reset the election timeout"
         _logger.info(c["raft_logic"] + "Node ID: %d -- Resetting election timeout" + c["end_color"], self._node.id)
         loop = asyncio.get_event_loop()
-        self._next_election_timeout = loop.time() + self._election_timeout
+        """
+        When on_election_timeout is called, the state is changed to candidate and a new election is started by calling start_election function. 
+        In the start_election function, the election timeout is reset using self._next_election_timeout = loop.time() + self._election_timeout. 
+        If loop.time() returns a value greater than or equal to self._next_election_timeout, then the election timeout will be set to a time in the past, 
```

Comment:
Maybe you are right. But when *candidate 42* sent a request with a higher term to *node 43*, then it reset the timer and instantly ran the `_on_election_timeout()` method, which started the election for *candidate 43*. The instant call [in this line](https://github.com/OpenCyphal-Garage/cyraft/pull/11/files#:~:text=self._election_timer%20%3D,self._on_election_timeout())) can happen because of the characteristics of the `loop.call_at` method, since if we specify a later or negative time, it runs `self._on_term_timeout()` immediately. That's why I decided to use `loop.call_later`, since it configures the period after which the method is run, and not just configures the time to run the method.

Reply:
> call [in this line](https://github.com/OpenCyphal-Garage/cyraft/pull/11/files#:~:text=self._election_timer%20%3D,self._on_election_timeout()) can happen

Link doesn't work: https://github.com/OpenCyphal-Garage/cyraft/pull/11/files#:~:text=self._election_timer%20%3D,self._on_election_timeout()

It should look something like this: https://github.com/Parsifal22/cyraft/blob/67a48aa8fb28c0c0b070733da4da338e747eb738/cyraft/node.py#L702

<img width="575" alt="image" src="https://github.com/user-attachments/assets/f6ed72e8-6883-4b81-8fc9-bd6949eecc9a">

Reply:
If instead of using `call_at`, you'd like to use `call_later`, that's okay, but functionally it should do exactly the same:

<img width="832" alt="image" src="https://github.com/user-attachments/assets/f8d7fcb2-8cbf-423e-8410-4729771c5c53">

https://docs.python.org/3/library/asyncio-eventloop.html#asyncio.loop.call_at

As far as I can tell:

```py
def _reset_election_timeout(self) -> None:
        """
        If a follower receives a heartbeat from the leader, it should reset its election timeout.
        """
        assert self._state == RaftState.FOLLOWER, "Only followers should reset the election timeout"
        _logger.info(c["raft_logic"] + "Node ID: %d -- Resetting election timeout" + c["end_color"], self._node.id)
        loop = asyncio.get_event_loop()
        self._next_election_timeout = loop.time() + self._election_timeout
        if hasattr(self, "_election_timer"):
            self._election_timer.cancel()
        self._election_timer = loop.call_at(
            self._next_election_timeout, lambda: asyncio.create_task(self._on_election_timeout())
        )
```

Is the same as this:

```py
def _reset_election_timeout(self) -> None:
        """
        If a follower receives a heartbeat from the leader, it should reset its election timeout.
        """
        assert self._state == RaftState.FOLLOWER, "Only followers should reset the election timeout"
        _logger.info(c["raft_logic"] + "Node ID: %d -- Resetting election timeout" + c["end_color"], self._node.id)
        loop = asyncio.get_event_loop()
        if hasattr(self, "_election_timer"):
            self._election_timer.cancel()
        self._election_timer = loop.call_later(
            self._election_timeout, lambda: asyncio.create_task(self._on_election_timeout())
```

So you can keep this.